Armadillo/src/main/java/com/scribd/armadillo/error/ArmadilloException.kt

```diff
@@ -2,84 +2,103 @@ package com.scribd.armadillo.error
 
 import com.scribd.armadillo.actions.Action
 
-sealed class ArmadilloException(exception: Exception) : Exception(exception) {
+sealed class ArmadilloException(cause: Throwable? = null,
+                                message: String = "Unhandled Armadillo Error")
+    : Exception(message, cause) {
     abstract val errorCode: Int
 }
 
 /**
  * Errors in Armadillo
  */
-data class EngineNotInitialized(val reason: String) : ArmadilloException(exception = Exception(reason)) {
+class EngineNotInitialized(message: String)
+    : ArmadilloException(message = "Armadillo Engine is not initialized: $message") {
     override val errorCode = 100
 }
 
-object TransportControlsNull : ArmadilloException(Exception()) {
+class TransportControlsNull
+    : ArmadilloException(message = "Internal error, the transport controls are null. Are the controls being initialized?") {
     override val errorCode = 101
 }
```

Comment:
Similarly keep these as objects as there is no constructor param?
```suggestion
object TransportControlsNull : ArmadilloException(message = "Internal error, the transport controls are null. Are the controls being initialized?") {
    override val errorCode: Int = 101
    private fun readResolve(): Any = this
}
```

Reply:
I believe using `object` there is causing bugs and interfering with `stacktrace` which can cause annoyances.

Reply:
haa ok..